scripts/tests/run_tv_casting_test.py

```diff
@@ -0,0 +1,305 @@
+                        logging.info('Linux tv-app is up and running!')
+
+                        # If the string is found, then break out of the loop and go ahead with running the Linux tv-casting-app.
+                        break
+
+                # Open and write to the log file for the Linux tv-casting-app.
+                with LogFileManager(linux_tv_casting_app_log_path, 'w') as linux_tv_casting_app_log_file:
+                    tv_casting_app_abs_path = os.path.abspath(tv_casting_app_rel_path)
+
+                    # Run the Linux tv-casting-app subprocess.
+                    with ProcessManager(disable_stdout_buffering_cmd + [tv_casting_app_abs_path], stdout=subprocess.PIPE, stderr=subprocess.PIPE) as tv_casting_app_process:
+                        # Initialize variables.
+                        continue_parsing = False
+                        valid_discovered_commissioner = ''
+
+                        # Read the output as we receive it from the tv-casting-app subprocess.
+                        for line in tv_casting_app_process.stdout:
+                            # Write to the Linux tv-casting-app log file.
+                            linux_tv_casting_app_log_file.write(line)
+                            linux_tv_casting_app_log_file.flush()
+
+                            # Fail fast if "No commissioner discovered" string found.
+                            if "No commissioner discovered" in line:
+                                logging.error(line.rstrip('\n'))
+                                handle_discovery_failure([linux_tv_app_log_path, linux_tv_casting_app_log_path])
+
+                            # Look for 'Discovered Commissioner'.
+                            if "Discovered Commissioner" in line:
+                                valid_discovered_commissioner = line.rstrip('\n')
+
+                                # Continue parsing the content that belongs to the "Discovered Commissioner".
+                                continue_parsing = True
+
+                                # Initialize variables to store the information of interest.
+                                valid_vendor_id: Optional[str] = None
+                                valid_product_id: Optional[str] = None
+                                valid_device_type: Optional[str] = None
+
+                            if continue_parsing:
+                                # Check if the Vendor ID, Product ID, and Device Type match the expected constant values.
+                                # If they do not match, then handle the discovery failure.
+                                if 'Vendor ID:' in line:
+                                    valid_vendor_id = validate_value(VENDOR_ID, line, 'Vendor ID')
+
+                                    if not valid_vendor_id:
+                                        handle_discovery_failure([linux_tv_app_log_path, linux_tv_casting_app_log_path])
```

Comment:
Indents here are too deep. I cannot follow such a large and indented method easily. Could you split it in separate functions so that things are easier read?

Reply:
Sure, separate functions have been created in the latest commit.